test/e2e/testing-manifests/storage-csi/gce-pd/controller_service.yaml

```diff
@@ -0,0 +1,13 @@
+kind: Service
```

Comment:
Is a service necessary here? Nothing in the container should be calling our pods outside the csi.sock right?

Reply:
True, was vestigial from the deployment I was modeling from

Reply:
nevermind, StatefulSet's require a headless service. Adding it back in now.

Reply:
ports are also required. But from what I understand it just fowards port 12345 to the pods port 12345 which I never opened anyway. So I don't think its exposing anything it shouldn't be...

Reply:
Are you sure you need a service? I was able to create a statefulset without a service

Reply:
https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations
"StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service."

It seems that I am able to have a statefulset that works without a service but I still have to specify a "serviceName" in the statefulset. Without this it wont pass validation. But also the stateful set seems to work just fine without the service that the "serviceName" refers to. I'm open to having or not having the service, my worry would be that since the service is "technically required" it might break this test on a future version of Kubernetes when they decide to be more strict about it.

Reply:
Yuck, two bad choices. I'm fine either way then.

Reply:
Leaving the "dummy service" in then.